storage/statedb/database.go

```diff
@@ -822,9 +829,22 @@ func (db *Database) Commit(node common.Hash, report bool) error {
 	db.gcnodes, db.gcsize, db.gctime = 0, 0, 0
 	db.flushnodes, db.flushsize, db.flushtime = 0, 0, 0
 
+	if db.settingMigrationDBRequired(blockNum) {
+		db.diskDB.SetStateTrieMigrationDB(blockNum)
+	}
 	return nil
 }
 
+func (db *Database) settingMigrationDBRequired(blockNum uint64) bool {
+	if blockNum == 0 {
```

Comment:
Shouldn't `dataArchivingBlockNum` be compared to zero instead `blockNum`?

Reply:
I need to make sure `blockNum` is the same as `NoDataArchivingPreparation`.

Reply:
Updated!